### PR TITLE
feature: time periods support for API level response status analytics

### PR DESCRIPTION
**Pull request by JedrzejJanasiak**
## Issue

https://gravitee.atlassian.net/browse/APIM-7414

## Description

Backend part for supporting different time ranges for API level response status analytics 

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-elcyctgaso.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/resource/api/analytics/ApiAnalyticsResource.java [hunk 1]
```diff
@@ -115,8 +115,10 @@ public ApiAnalyticsAverageConnectionDurationResponse getAverageConnectionDuratio
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     @Permissions({ @Permission(value = RolePermission.API_ANALYTICS, acls = { RolePermissionAction.READ }) })
-    public ApiAnalyticsResponseStatusRangesResponse getResponseStatusRanges() {
-        var request = new SearchResponseStatusRangesUseCase.Input(apiId, GraviteeContext.getCurrentEnvironment());
+    public ApiAnalyticsResponseStatusRangesResponse getResponseStatusRanges(@QueryParam("from") Long from, @QueryParam("to") Long to) {
+        long end = to != null ? to : Instant.now().toEpochMilli();
```
Comment (jgiovaresco) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/resource/api/analytics/ApiAnalyticsResource.java]: This (and next line) should be in the usecase. It is part of the business rules and there should be tested at the use case level

**Review by jgiovaresco — CHANGES_REQUESTED** [review]
The PR is incomplete we need to update the ElasticSearch query to filter using time period provided (https://github.com/gravitee-io/gravitee-api-management/blob/master/gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/analytics/adapter/SearchResponseStatusRangesQueryAdapter.java)

Check https://github.com/gravitee-io/gravitee-api-management/pull/9800

**Comment by JedrzejJanasiak** [comment]
> The PR is incomplete we need to update the ElasticSearch query to filter using period provided (https://github.com/gravitee-io/gravitee-api-management/blob/master/gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/analytics/adapter/SearchResponseStatusRangesQueryAdapter.java)
> 
> Check #9800

We did, there is a method on lines 96-101 ;) The query code is shared between the dashboard and API-level analytics so the date filtering should be ok. There are tests to cover that. I just had to add _from_ and _to_ parameters to the API level use case :)

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/resource/api/analytics/ApiAnalyticsResource.java (hunk 1 above)
Comment (JedrzejJanasiak) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/resource/api/analytics/ApiAnalyticsResource.java]: done

**Review by jgiovaresco — APPROVED** [review]

**Pull request merged**